crates/ef-testing-proc-macro/src/content_reader.rs

```diff
@@ -0,0 +1,54 @@
+use std::collections::BTreeMap;
+
+use serde_json::Value;
+
+use crate::{dir_reader::PathWrapper, utils::blockchain_tests_to_general_state_tests_path};
+
+pub struct ContentReader;
+
+impl ContentReader {
+    pub fn secret_key(path: PathWrapper) -> Result<Value, eyre::Error> {
+        let path = blockchain_tests_to_general_state_tests_path(path);
+        let content = path.read_file_to_string()?;
+
+        let cases: BTreeMap<String, Value> = serde_json::from_str(&content)?;
+        let case = cases
+            .into_values()
+            .collect::<Vec<_>>()
+            .first()
+            .ok_or_else(|| eyre::eyre!("No case found"))?
+            .clone();
```

Comment:
how about
`let case = cases.into_values().next().ok_or_else(|| eyre::eyre!("No case found"))?;`

Reply:
I think that works, nice